src/plugin.ts

```diff
@@ -58,8 +58,37 @@ export class Plugin {
     this.pluginPath = path;
 
     this.version = packageJSON.version || "0.0.0";
-    this.main = packageJSON.main || "./index.js"; // figure out the main module - index.js unless otherwise specified
     this.isESM = packageJSON.type === "module";
+    this.main = "";
+
+    // figure out the main module
+    // exports is available - https://nodejs.org/dist/latest-v14.x/docs/api/packages.html#packages_package_entry_points
+    if (packageJSON.exports) {
+      // main entrypoint - https://nodejs.org/dist/latest-v14.x/docs/api/packages.html#packages_main_entry_point_export
+      if (typeof packageJSON.exports === "string") {
+        this.main = packageJSON.exports;  
+      } else { // subpath export - https://nodejs.org/dist/latest-v14.x/docs/api/packages.html#packages_subpath_exports
+        // conditional exports - https://nodejs.org/dist/latest-v14.x/docs/api/packages.html#packages_conditional_exports
+        const exports = packageJSON.exports.import || packageJSON.exports.require || packageJSON.exports.node || packageJSON.exports.default || packageJSON.exports["."];
+
+        // check if conditional export is nested
+        if (typeof exports !== "string") {
+          if(exports.import) {
+            this.main = exports.import;
+            this.isESM = true;
```

Comment:
That's not at all how Node.js does it, `isESM` should be equals to `this.main.endsWith('.mjs') || (this.main.endsWith('.js') && packageJSON.type === "module")`.

Reply:
I already do the packageJSON type check in line 61.
I could probably move the isESM check after the if opened on line 66. But are you sure that `exports.import` does not automatically assume ESM?

Reply:
I'm sure
```sh
mkdir repro
cd repro
echo '{"name":"myPackage","exports":{"import":"./entry.cjs"}}' > package.json
echo 'console.log(this === undefined ? "ESM" : "CJS")' > entry.cjs
node --input-type=module -e 'import "myPackage"'
cd ..
rm -r repro
```